MIGRATION_GUIDE.md

```diff
@@ -76,6 +77,27 @@ func (m MyCustomClaims) Validate() error {
 }
 ```
 
+## Changes to the `Token` and `Parser` struct
+
+The previously global functions `DecodeSegment` and `EncodeSegment` were moved to the `Parser` and `Token` struct respectively. This will allow us in the futureu to configure the behavior of these two based on options supplied on the parser or the token (creation). This also removes two previously global variables and moves them to parser options `WithStrictDecoding` and `WithPaddingAllowed`.
```

Comment:
```suggestion
The previously global functions `DecodeSegment` and `EncodeSegment` were moved to the `Parser` and `Token` struct respectively. This will allow us in the future to configure the behavior of these two based on options supplied on the parser or the token (creation). This also removes two previously global variables and moves them to parser options `WithStrictDecoding` and `WithPaddingAllowed`.
```

^ Small (subjective) nit, wrapped text facilitates review.  We could cap the wrapping column to 100 for markdown 

This has been on my mind for other reason FYI, just figured I'd drop this as an idea to simplify our life.

https://mtsknn.fi/blog/4-1-wrapping-styles-for-markdown-prose-and-code-comments/#benefits-of-a-consistent-wrapping-style
https://about.gitlab.com/blog/2016/10/11/wrapping-text/

Reply:
Ah yes, good idea. I always try to wrap the documentation inside our code at either 80 or 100 lines anyway, so having this for markdown as well makes sense.